src/FlightMap/Widgets/QGCInstrumentWidget.qml

```diff
@@ -55,6 +59,111 @@ ColumnLayout {
             vehicle:                globals.activeVehicle
             anchors.verticalCenter: parent.verticalCenter
         }
+    }  
+
+    enum WinchCommands {
+        RELEASE = 0,
+        DELIVER = 4,
+        RETRACT = 6
+    }
+    
+    property int _currentCommand:   -1
+    property int _WinchNodeID:      42
+    property int _MAVLink_CMD_ID:   42600
+
+    QGCButton {
+        id:                         winchEmergency
+        backRadius:                 10
+        showBorder:                 true
+        height:                     _outerRadius
+        Layout.fillWidth:           true
+        Layout.topMargin:           Window.height - 370
+        text:                       "Emergency release"
+        background:                 Rectangle {
+            color:                  "red"
+            radius:                 10
+            border.color:           "white"
+            border.width:           1
+        }
+        checkable:                  true
+        Image {
+            id:                     alert
+            source:                 "/qmlimages/Yield.svg"
+            height:                 parent.height
+            width:                  height
+            x:                      10
+        }
+        onClicked:  {
+            _currentCommand = QGCInstrumentWidget.WinchCommands.RELEASE;
+            winchDeliver.checked = winchRetract.checked = false;
+            if (checked) {slider.visible = true; slider.confirmText = qsTr("Emergency release winch")}
+            else {slider.visible = false}
+        }
```

Comment:
Strange formatting, whitespace and indentation make this hard to read.

Also, including javascript statements directly is somewhat confusing. A nicer solution IMO is to define a named javascript function below, and then just say onClicked: myFunction

```suggestion
        onClicked:  releaseWinch
        
        function releaseWinch() {
            _currentCommand = QGCInstrumentWidget.WinchCommands.RELEASE;
            winchDeliver.checked = false;
            winchRetract.checked = false;
            if (checked) {
                slider.visible = true;
                slider.confirmText = qsTr("Emergency release winch");
            } else {
                slider.visible = false;
            }
        }
```

Reply:
Follow-up after reading the rest of the code: Each of the buttons do exactly the same thing, except that the text and command is different. Consider reusing a function to reduce complexity.

Reply:
Moved code to functions like suggested and improved formatting. I decided not to use just one function for all three behaviors for two reasons:

1. The command, text string and what buttons to uncheck is unique for each function. I could pass these as arguments or switch inside the function, but it looked messy IMO.
2. The explicit names make the flow easier to follow IMO.